### ISSUE TITLE
Elastic search 5.2.2 image, tcp service binds to 127.0.0.1

### ISSUE THREAD

**Issue by jCalamari** [elastic/elasticsearch-docker#34]
## Bug description

Tcp service binds to 127.0.0.1 and makes it inaccessible from outside of the container. Http service binds to 0.0.0.0

Elastic search log:

```
elasticsearch    | [2017-03-10T15:35:32,418][WARN ][i.n.u.i.MacAddressUtil   ] Failed to find a usable hardware address from the network interfaces; using random bytes: 17:75:0b:37:b3:dd:79:c6
elasticsearch    | [2017-03-10T15:35:32,532][INFO ][o.e.t.TransportService   ] [fye6pJB] publish_address {127.0.0.1:9300}, bound_addresses {[::1]:9300}, {127.0.0.1:9300}
elasticsearch    | [2017-03-10T15:35:35,630][INFO ][o.e.c.s.ClusterService   ] [fye6pJB] new_master {fye6pJB}{fye6pJB4SgO2oz4NQ0UKvg}{kF0EEapzQaWTIZARDM5CDA}{127.0.0.1}{127.0.0.1:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
elasticsearch    | [2017-03-10T15:35:35,656][INFO ][o.e.h.HttpServer         ] [fye6pJB] publish_address {172.18.0.2:9200}, bound_addresses {[::]:9200}
elasticsearch    | [2017-03-10T15:35:35,657][INFO ][o.e.n.Node               ] [fye6pJB] started
elasticsearch    | [2017-03-10T15:35:35,704][INFO ][o.e.g.GatewayService     ] [fye6pJB] recovered [0] indices into cluster_state
```

docker-info:

```
Containers: 4
 Running: 3
 Paused: 0
 Stopped: 1
Images: 321
Server Version: 17.03.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 977c511eda0925a723debdc94d09459af49d082a
runc version: a01dafd48bc1c7cc12bdb01206f9fea7dd6feb70
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.12-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.65 GiB
Name: moby
ID: 2XJR:H7L7:UVXN:VAWH:2EHG:GPUZ:GQSY:I55R:XJUA:3LUX:4VH5:6UMC
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 45
 Goroutines: 64
 System Time: 2017-03-10T15:37:54.026016483Z
 EventsListeners: 2
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
````

docker-compose version:

```
docker-compose version 1.11.2, build dfed245
docker-py version: 2.1.0
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
```

uname -a:

```
Darwin 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64 x86_64
```

**Comment by jCalamari** [comment]
> Note: since 5.0, Elasticsearch only listens on localhost by default on both http and transport, so this image sets http.host to 0.0.0.0 (given that localhost is not terribly useful in the Docker context).

Sorry for my ignorance. Closing.

**Comment by dliappis** [comment]
@jCalamari I am glad you sorted it out. Also note that the image maintained by Elastic is hosted in a dedicated registry and needs to be pulled as described [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html).

**Comment by giorgiovinci** [comment]
I'm trying to quickly start a local instance of elasticsearch without really doing too much configuration.
Unfortunately I do need the 9300 port binded on my local machine so I'm trying all kind of parameters on my docker command but nothing seems to work. From your comment I tried:
docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" -e"http.host=0.0.0.0" elasticsearch:5.6.16
Without success.
I've also tried "transport.host=0.0.0.0" but nothing seems to work.
I'm clearly missing something here but I'm flabbergasted that there's no simple/clear answer in the web about this.
